logicle/app/chat/assistants/mine/page.tsx

```diff
@@ -39,8 +40,25 @@ const MyAssistantPage = () => {
   const modalContext = useConfirmationContext()
   const [searchTerm, setSearchTerm] = useState<string>('')
 
-  const isMine = (assistant, profile) => {
-    return assistant.owner == profile?.id
+  /**
+   * @param assistant 
+   * @param profile the user profile
+   * @returns whether the user can edit the assistant
+   */
+  const canEditAssistant = (assistant: dto.UserAssistant, profile: dto.UserProfile | undefined) => {
+    // A user can edit the assistant if:
+    // - he is the owner
+    // - he has the WorkspaceRole Editor role in the same workspace where the assistant has been shared
+    //   (if the assistant has been shared to all it is editable only by the owner)
+    if (assistant.owner == profile?.id) return true;
+
+    return assistant.sharing.some(s => {
+      if (dto.isAllSharingType(s)) return false;
+
+      return profile?.workspaces.some(w => {
+        return w.id == s.workspaceId && w.role == WorkspaceRole.EDITOR
```

Comment:
Admin and Owner must be able to modify the assistant.
As there's no multiple workspace role support, higher level roles (OWNER -> ADMIN -> EDITOR -> MEMBER) include privileges from lower level roles.